pkg/config/config.go

```diff
@@ -0,0 +1,303 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/iver-wharf/wharf-core/pkg/config"
+	v1 "k8s.io/api/core/v1"
+)
+
+// Config holds all configurable settings for wharf-api.
+//
+// The config is read in the following order:
+//
+// 1. File: /etc/iver-wharf/wharf-cmd/wharf-cmd-config.yml
+//
+// 2. File: ./wharf-cmd-config.yml
+//
+// 3. File from environment variable: WHARF_CONFIG
+//
+// 4. Environment variables, prefixed with WHARF_
+//
+// Each inner struct is represented as a deeper field in the different
+// configurations. For YAML they represent deeper nested maps. For environment
+// variables they are joined together by underscores.
+//
+// All environment variables must be uppercased, while YAML files are
+// case-insensitive. Keeping camelCasing in YAML config files is recommended
+// for consistency.
+type Config struct {
+	K8s         K8sConfig
+	Worker      WorkerConfig
+	Provisioner ProvisionerConfig
+	Aggregator  AggregatorConfig
+	Watchdog    WatchdogConfig
+}
+
+// K8sConfig holds settings for using kubernetes.
+type K8sConfig struct {
+	// Context is the context used when talking to kubernetes.
+	//
+	// Added in v0.8.0.
+	Context string
+	// Namespace is the kubernetes namespace to talk to.
+	//
+	// Added in v0.8.0.
+	Namespace string
+}
+
+// WorkerConfig holds settings for the worker.
+type WorkerConfig struct {
+	Steps StepsConfig
+}
+
+// StepsConfig holds settings for the different types of steps.
+type StepsConfig struct {
+	Docker  DockerStepConfig
+	Kubectl KubectlStepConfig
+	Helm    HelmStepConfig
```

Comment:
nit: For completeness sake, could you also add configs already for the other step type's Docker images?

Alt. add TODOs that they need to be extracted to configs

Reply:
Going with alternative, adding TODOs.